routes/plant/plantprofile.js

```diff
@@ -0,0 +1,58 @@
+const express = require('express');
+const router = express.Router();
+
+// Import schemas
+const User = require('../../schemas/user');
+const Plant = require('../../schemas/plant');
+
+// Authenticate user
+const authenticateUser = async (req, res, next) => {
+    const accessToken = req.header("Authorization");
+    try {
+      const user = await User.findOne({accessToken: accessToken});
+      if (user) {
+        next();
+      } else {
+        res.status(401).json({
+          success: false,
+          response: "Please log in"
+        })
+      } 
+    } catch (e) {
+      res.status(500).json({
+        success: false,
+        response: e
+      })
+    }
+  };
+
+// Plant profile
+router.get("/:username/garden/:plantId", authenticateUser);
```

Comment:
In deleteplants.js & editpants.js the route is 
":username/garden/:plantId"
just like it is here in plantprofile.js,  

while in garden.js is "/users/:username/garden"